### ISSUE TITLE
Istio sidecar injection fails on scale down in EKS

### ISSUE THREAD

**Issue by ljdev92** [istio/istio#13762]
**Bug description**
On a scale down event in the EKS ASG, whether via the cluster autoscaler or after forcibly terminating a node, the api server can fail to talk to the sidecar injector service, throwing errors such as:

`Failed calling webhook, failing closed sidecar-injector.istio.io: failed calling admission webhook "sidecar-injector.istio.io": Post https://istio-sidecar-injector.istio-system.svc:443/inject?timeout=30s: context deadline exceeded (Client.Timeout exceeded while awaiting headers)`

This leads to a failure in rescheduling the pods from the scaled down nodes which require the sidecar.

**Expected behaviour**
Sidecar should be injected into the rescheduled pods

**Version**
Istio version 1.1.3
EKS 1.12.6
EKS Node 1.12.7

**How was Istio installed?**
Helm

**Environment where bug was observed (cloud vendor, OS, etc)**
EKS

Any help would be greatly appreciated

**Comment by mikebryant** [comment]
Is this a transient failure? i.e. it happens immediately after the scale down event, but recovers on it's own?

(I'm wondering, are you only running one copy of the sidecar-injector, and it's on the node that gets terminated, so it takes time to be redeployed onto a different node)

**Comment by ljdev92** [comment]
@mikebryant It doesn't seem to recover at all, once the replica set controller fails to call the sidecar injector it gives up trying.

We've tried running multiple sidecar-injector pods using anti-affinity rules to make sure they're on separate nodes but the problem still occurs. We've switched to manual injection as a temporary solution but automatic injection would be preferable.

**Comment by pc-rshetty** [comment]
I see this error even when doing rolling upgrade of worker nodes

**Comment by sobolevan** [comment]
I have the same issue as @pc-rshetty

**Comment by pc-rshetty** [comment]
I had logged a defect for this https://github.com/istio/istio/issues/13840 
Only difference i see is in case of rolling upgrade of worker nodes the pods disappear and then come back up after 17+ mins

**Comment by pc-rshetty** [comment]
Also Istio and EKS are the least common denominators

**Comment by howardjohn** [comment]
It seems to me the issue is there is no PDB for the injector, so it will get scaled to 0 and block other pods from starting up. In https://github.com/istio/istio/pull/14210 we added a PDB which I suspect would fix it

**Comment by markuszm** [comment]
We have the same issue as @pc-rshetty 
I tried running the sidecar-injector as a DaemonSet to ensure that there is always a sidecar-injector pod is running and we are running the istio pilot HA as well. Still, when doing a rolling upgrade of the autoscaling group via Cloudformation or when forcibly terminating a node, there are cases where even when a sidecar-injector is still running on a remaining node there is a timeout on the pod initialization in the replica set. We tried to change the log level of the injector to debug but no admission requests that fail are seen in the logs. 

Could there be an issue with EKS where the worker nodes and control plane nodes have issues to communicate as the webhook is coming from the control plane? 
We see the same behavior that after around 15 min the pods are scheduled as the webhook does not timeout anymore.
